pkg/skaffold/test/custom/custom.go

```diff
@@ -32,34 +36,42 @@ import (
 	"github.com/GoogleContainerTools/skaffold/pkg/skaffold/util"
 )
 
-// for tests
-var doRunCustomCommand = runCustomCommand
+var (
+	// for tests
+	testContext = retrieveTestContext
+
+	// Image is an environment variable key, whose value is the
+	// fully qualified image name passed in to a custom command.
+	Image = "IMAGE"
+)
 
 const Windows string = "windows"
 
 type Runner struct {
 	customTest     latest.CustomTest
+	imageName      string
 	testWorkingDir string
 }
 
 // New creates a new custom.Runner.
-func New(cfg docker.Config, wd string, ct latest.CustomTest) (*Runner, error) {
+func New(cfg docker.Config, imageName string, wd string, ct latest.CustomTest) (*Runner, error) {
 	return &Runner{
+		imageName:      imageName,
 		customTest:     ct,
 		testWorkingDir: wd,
 	}, nil
 }
 
 // Test is the entrypoint for running custom tests
-func (ct *Runner) Test(ctx context.Context, out io.Writer, _ []build.Artifact) error {
-	if err := doRunCustomCommand(ctx, out, ct.customTest); err != nil {
+func (ct *Runner) Test(ctx context.Context, out io.Writer, bRes []build.Artifact) error {
+	if err := ct.runCustomCommand(ctx, out, ct.customTest, bRes); err != nil {
```

Comment:
`ct.customTest` is already a field in `ct` so why does it need to be a function parameter in `runCustomCommand` also?

Reply:
This wan not a member method before. I just changed it. Removed the test param.